lib/services/local/integrations.go

```diff
@@ -173,6 +197,41 @@ func notReferencedByEAS(ctx context.Context, bk backend.Backend, name string) ([
 	return conditionalActions, nil
 }
 
+// integrationReferencedByAWSICPlugin returns an error if the integration name is referenced
+// by an existing AWS Identity Center plugin. In case the AWS Identity Center plugin exists
+// but does not reference this integration, a conditional action is returned with a revision
+// of the plugin to ensure that plugin hasn't changed during deletion of the AWS OIDC integration.
+// Returns nil ConditionalAction if AWS Identity Center plugin is not found.
+func integrationReferencedByAWSICPlugin(ctx context.Context, bk backend.Backend, name string) (*backend.ConditionalAction, error) {
+	pluginService := NewPluginsService(bk)
+	plugins, err := pluginService.GetPlugins(ctx, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	for _, p := range plugins {
+		pluginV1, ok := p.(*types.PluginV1)
+		if !ok {
+			continue
+		}
+
+		if pluginV1.GetType() == types.PluginType(types.PluginTypeAWSIdentityCenter) {
```

Comment:
Nit: Use switch.

Reply:
Kept it as is. The inner block already uses switch and using switch in both places read confusing. 
```
if pluginV1.GetType() == types.PluginType(types.PluginTypeAWSIdentityCenter) {
	switch pluginV1.Spec.GetAwsIc().IntegrationName {
	case name:
		return nil, trace.BadParameter("cannot delete AWS OIDC integration currently referenced by AWS Identity Center integration %q", pluginV1.GetName())
	default:
		conditionalActions = append(conditionalActions, backend.ConditionalAction{
			Key:       backend.NewKey(pluginsPrefix, name),
			Action:    backend.Nop(),
			Condition: backend.Revision(pluginV1.GetRevision()),
		})
		return conditionalActions, nil
	}
		}
```